src/main/AuthenticationRequest.ts

```diff
@@ -248,66 +252,44 @@ function assertValidRequestOpts(opts: AuthenticationRequestOpts) {
-  if (!opts || !opts.presentationDefinitions || opts.presentationDefinitions.length == 0) {
+function createClaimsPayload(opts: ClaimOpts): ClaimPayload {
+  if (!opts || (!opts.vpToken.presentationDefinition && !opts.vpToken?.presentationDefinitionUri)) {
     return undefined;
   }
-  let vp_token: VpTokenClaimPayload;
-  let id_token: IdTokenClaimPayload;
   const pex: PEX = new PEX();
-  opts.presentationDefinitions.forEach((def) => {
-    const discoveryResult = pex.definitionVersionDiscovery(def.definition);
-    if (discoveryResult.error) {
-      throw new Error(SIOPErrors.REQUEST_CLAIMS_PRESENTATION_DEFINITION_NOT_VALID);
-    }
-    switch (def.location) {
-      case PresentationLocation.ID_TOKEN: {
-        if (!id_token || !id_token.verifiable_presentations) {
-          id_token = { verifiable_presentations: [{ presentation_definition: def.definition }] };
-        } else {
-          id_token.verifiable_presentations.push({ presentation_definition: def.definition });
-        }
-        return;
-      }
-      case PresentationLocation.VP_TOKEN: {
-        if (vp_token) {
-          // There can only be one definition in the vp_token according to the spec
-          throw new Error(SIOPErrors.REQUEST_CLAIMS_PRESENTATION_DEFINITION_NOT_VALID);
-        } else {
-          vp_token = {
-            nonce: nonce,
-            presentation_definition: def.definition,
-            response_type: PresentationLocation.VP_TOKEN,
-          };
-        }
-        return;
-      }
-    }
-  });
-  const payload: ClaimPayload = {
-    id_token,
-    vp_token,
+  const discoveryResult = pex.definitionVersionDiscovery(opts.vpToken.presentationDefinition);
+  if (discoveryResult.error) {
+    throw new Error(SIOPErrors.REQUEST_CLAIMS_PRESENTATION_DEFINITION_NOT_VALID);
+  }
+
+  return {
+    id_token: opts.idToken,
+    vp_token: {
+      ...(opts.vpToken.presentationDefinition ? { presentation_definition: opts.vpToken.presentationDefinition } : {}),
+      ...(opts.vpToken.presentationDefinitionUri ? { presentation_definition_uri: opts.vpToken.presentationDefinitionUri } : {}),
+    },
   };
-  return payload;
 }
 
 async function createAuthenticationRequestPayload(opts: AuthenticationRequestOpts): Promise<AuthenticationRequestPayload> {
   assertValidRequestOpts(opts);
   const state = getState(opts.state);
-  const registration = await createRequestRegistration(opts.registration);
-  const claims = createClaimsPayload(opts.claims, opts.nonce);
+  const registration = await createRequestRegistration(opts['registration']);
+  const claims = createClaimsPayload(opts.claims);
 
   return {
     response_type: ResponseType.ID_TOKEN,
     scope: Scope.OPENID,
-    client_id: opts.signatureType.did || opts.redirectUri, //todo: check whether we should include opts.redirectUri value here, or the whole of client_id to begin with
+    client_id: opts.signatureType.did || opts['registrationUri'],
```

Comment:
Some confusion with the text:

**If client_id is a HTTPS URL, client_id is resolved to obtain all Relying Party metadata from an Entity Statement as defined in [[OpenID.Federation](https://openid.net/specs/openid-connect-self-issued-v2-1_0-ID1.html#OpenID.Federation)].** 

The text is talking about the OP side where the client id (HTTPS URL) is resolved and the code above is in the RP side where we just need to set it. Obviously this is neither registration nor registrationUri but it cannot be the redirectUri either because it returns a JWT with the relying party metadata, so it needs to come from registration.client_id. 

According to the text [Automatic Registration](https://openid.net/specs/openid-connect-federation-1_0.html#name-automatic-registration) and [Federation Entity Configuration](https://openid.net/specs/openid-connect-federation-1_0.html#name-obtaining-federation-entity) with automatic registration (registration and authentication in a single request) client_id is a /.well-know/openid-federation url from where the RP configuration can be retrieved. I think we need a TODO in this case.

Reply:
Client id is used for app/client auth
The client id for now needs to be something which is provided by the OP as an argument. Either you will get a dynamic version when using dynamic registration, or you are using a static version as it was given/handed out by the AS/RP beforehand.

Reply:
Fixed!